pkg/server/api_registration.go

```diff
@@ -159,33 +163,20 @@ func (i *InventoryServer) writeMachineInventoryCloudConfig(conn *websocket.Conn,
 		registration.Spec.Config = &elementalv1.Config{}
 	}
 
-	elementalConf := elementalv1.Elemental{
-		Registration: elementalv1.Registration{
-			URL:    registration.Status.RegistrationURL,
-			CACert: i.getRancherCACert(),
-		},
-		SystemAgent: elementalv1.SystemAgent{
-			URL:             fmt.Sprintf("%s/k8s/clusters/local", serverURL),
-			Token:           string(secret.Data["token"]),
-			SecretName:      inventory.Name,
-			SecretNamespace: inventory.Namespace,
-		},
-		Install: registration.Spec.Config.Elemental.Install,
-		Reset:   registration.Spec.Config.Elemental.Reset,
+	config, err := registration.GetClientRegistrationConfig(i.getRancherCACert())
```

Comment:
here we get just the Elemental.Registration config now, which is the one we return for unauthenticated requrest, the one to be put in the ISO.
We are supposed to pass the full config here, in particular the SystemAgent one.